benchexec/tools/cpa-witness-exec.py

```diff
@@ -0,0 +1,86 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+# prepare for Python 3
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+
+import benchexec.result as result
+import benchexec.util as util
+import benchexec.tools.cpachecker as cpachecker
+from benchexec.model import SOFTTIMELIMIT
+
+
+class Tool(cpachecker.Tool):
+    """
+    Tool info for CPA-witness-exec.
+    """
+
+    def executable(self):
+        super(Tool, self).executable()  # Makes sure that CPAchecker can be called, shows a warning otherwise
+        return util.find_executable('cpa_witness_exec.py', 'scripts/cpa_witness_exec.py')
+
+    def version(self, executable):
+        stdout = self._version_from_tool(executable, '-version')
+        return stdout
+
+    def name(self):
+        return 'CPA-Witness-Exec'
+
+    def cmdline(self, executable, options, tasks, propertyfile=None, rlimits={}):
+        added_options = []
+        if SOFTTIMELIMIT in rlimits:
+            if "-timelimit" in options:
+                logging.warning('Time limit already specified in command-line options,' +
+                                ' not adding time limit from benchmark definition to the command line.')
+            else:
+                added_options += ["-timelimit", str(rlimits[SOFTTIMELIMIT]) + 's']  # benchmark-xml uses seconds as unit
+
+        if "-stats" not in options:
+            added_options.append("-stats")
+
+        spec = ["-spec", propertyfile] if propertyfile is not None else []
+        added_options += spec
+        # Add additional options in front of existing ones, since -gcc-args ... must be last argument in front of task
+        return [executable] + added_options + options + tasks
+
+    def determine_result(self, returncode, returnsignal, output, isTimeout):
+        """
+        @param returncode: code returned by CPA-witness-exec
+        @param returnsignal: signal, which terminated CPA-witness-exec
+        @param output: the output of CPA-witness-exec
+        @return: status of CPA-witness-exec after executing a run
+        """
+
+        status = None
+        for line in reversed(output):
+            if line.startswith('Verification result: '):
+                line = line[21:].strip()
+                if line.startswith('FALSE'):
+                    status = result.RESULT_FALSE_REACH
+                else:
+                    status = result.RESULT_UNKNOWN
+                break
+
+        if not status:
+            status = result.RESULT_ERROR
```

Comment:
Can this module give more descriptive errors like CPAchecker's module does?

For example, IIRC in SV-Comp it was a requirement to detect and warn about invalid witnesses (cf. https://github.com/sosy-lab/benchexec/commit/363eed068ad0380c9744299525f70253415861d2). Cases like `exception` are useful for debugging.